src/Transport/Addressing/Partitioning/Strategies/SingleNamespacePartitioningStrategy.cs

```diff
@@ -7,32 +7,20 @@ namespace NServiceBus.AzureServiceBus.Addressing
 
     public class SingleNamespacePartitioningStrategy : INamespacePartitioningStrategy
     {
-        readonly string _connectionstring;
+        private readonly NamespacesDefinition _namespaces;
 
         public SingleNamespacePartitioningStrategy(ReadOnlySettings settings)
         {
-            List<string> connectionstrings;
-            if (!settings.TryGet(WellKnownConfigurationKeys.Topology.Addressing.Partitioning.Namespaces, out connectionstrings))
-            {
-                connectionstrings = new List<string>();
-            }
-
-            if (connectionstrings.Count == 0)
+            if (!settings.TryGet(WellKnownConfigurationKeys.Topology.Addressing.Partitioning.Namespaces, out _namespaces) || _namespaces.Count != 1)
             {
                 throw new ConfigurationErrorsException("The 'Single' namespace partitioning strategy requires exactly one namespace, please configure the connection string to your azure servicebus namespace");
             }
-
-            if (connectionstrings.Count > 1)
```

Comment:
The two checks have been joined into one.
Before they were:
`connectionStrings.Count == 0 or connectionStrings.Count > 1`

Now it's:
`connectionStrings.Count != 1`